librarian_packages/server/librarian_server/__init__.py

```diff
@@ -208,9 +208,13 @@ def commandline(argv):
             search.register_standing_order_checkin()
 
         # Hack the logger to indicate which server we are.
-        import tornado.process
-        fmtr = logging.getLogger('').handlers[0].formatter
-        fmtr._fmt = fmtr._fmt.replace(': ', ' #%d: ' % tornado.process.task_id())
+        try:
+            import tornado.process
+            fmtr = logging.getLogger('').handlers[0].formatter
+            fmtr._fmt = fmtr._fmt.replace(': ', ' #%d: ' % tornado.process.task_id())
+        except TypeError:
+            # task_id() returned None; don't change formatter
+            pass
```

Comment:
Stylistically, I try to keep my try/except blocks as minimal as possible to avoid potentially missing unexpected exceptions ... here, if somehow the `getLogger()` call raised a TypeError, you'd never know. In this case, based on the comment, the exceptional circumstance is very isolate-able:

```
taskid = tornado.process.task_id()
if taskid is not None:
   ... futz with loggger
```

Reply:
I've refactored as suggested.